### PR TITLE
Introduce a config setting to limit the number of tasks to be claimed

### PR DESCRIPTION
**Pull request by petemoore**
This makes integration testing easier (so tests can run worker to just pick up as many tasks as they have generated) and also makes it easier to customise worker behaviour from outside the worker (e.g. to execute arbitrary steps between each task).

### CODE REVIEW

File: worker/config.go [hunk 1]
```diff
@@ -238,13 +239,23 @@ func ConfigSchema() schematypes.Object {
...
 			},
+			"maxTasksToRun": schematypes.Integer{
+				MetaData: schematypes.MetaData{
+					Title: "Number of tasks the worker should run before exiting",
+					Description: `If set to 0, the worker does not limit the number of tasks
+						it will claim and execute. For positive values > 0, the worker will
+						exit if it completes the given number of tasks.`,
+				},
+				Minimum: 0,
+				Maximum: math.MaxInt32,
```
Comment (jonasfj) [inline worker/config.go]: Nit but no limit is signaled with `MaxInt64`

File: worker/queueservice.go [hunk 2]
```diff
@@ -85,10 +87,11 @@ func (q *queueService) Start() <-chan *taskClaim {
 
 	go func() {
 		for !q.halt.Get() {
-			q.mu.RLock()
-			capacity := q.capacity
-			q.mu.RUnlock()
-			tasks := q.retrieveTasksFromQueue(capacity)
+			claimSize, finished := q.claimSize()
+			if finished {
+				break
```
Comment (jonasfj) [inline worker/queueservice.go]: Shouldn't it just skip claiming? I might be wrong here... ie. Sleep a second and then `continue`.

Ideally, we probably should be doing Cond.Wait or something, but let's not complicate with that yet...

File: worker/config.go (hunk 1 above)
Comment (petemoore) [inline worker/config.go]: See https://golang.org/pkg/builtin/#int

File: worker/queueservice.go (hunk 2 above)
Comment (petemoore) [inline worker/queueservice.go]: We should break from the loop once we have claimed all of the tasks we are allowed to (i.e. once we hit `maxTasksToRun`).

File: worker/config.go (hunk 1 above)
Comment (petemoore) [inline worker/config.go]: See https://golang.org/pkg/builtin/#int

File: worker/queueservice.go (hunk 2 above)
Comment (petemoore) [inline worker/queueservice.go]: We should break from the loop once we have claimed all of the tasks we are allowed to (i.e. once we hit `maxTasksToRun`). At this point we know we cannot claim any more tasks, so no point looping.

**Pull request merged**